sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -411,12 +414,17 @@ private static String getRedactedUrl(URL url, Set<String> allowedQueryParameterN
      * @param headers HTTP headers on the request or response.
      * @param sb StringBuilder that is generating the log message.
      * @param logLevel Log level the environment is configured to use.
+     * @param disableRedactedHeaderLogging Flag indicating if redacted headers should be logged.
      */
     private static void addHeadersToLogMessage(Set<String> allowedHeaderNames, HttpHeaders headers,
-        LoggingEventBuilder logBuilder) {
+        LoggingEventBuilder logBuilder, boolean enableRedactedHeaderLogging) {
         // The raw header map uses keys that are already lower-cased.
-        HttpHeadersAccessHelper.getRawHeaderMap(headers).forEach((key, value) -> logBuilder.addKeyValue(value.getName(),
-            allowedHeaderNames.contains(key) ? value.getValue() : REDACTED_PLACEHOLDER));
+        HttpHeadersAccessHelper.getRawHeaderMap(headers).forEach((key, value) -> {
+            boolean isAllowed = allowedHeaderNames.contains(key);
+            if (isAllowed || enableRedactedHeaderLogging) {
+                logBuilder.addKeyValue(value.getName(), isAllowed ? value.getValue() : REDACTED_PLACEHOLDER);
+            }
```

Comment:
Should we group all redacted headers under one key-value pair?

`RedactedHeaders: header1, header2,...`

Reply:
we can do the mode when redacted headers are logged, I can add this